frontend/src/components/Portals/ScheduleModal.jsx

```diff
@@ -0,0 +1,376 @@
+                                        {(selectedTemplateId !== null && 
+                                            selectedDays.length !== 0 &&
+                                            !selectedStartTime.includes(null) &&
+                                            !selectedEndTime.includes(null)
+                                            ) 
+                                            ?
+                                            <button onClick={handleSubmit} className="bg-[#96d600] rounded-md p-2 pl-4 pr-4">Create rule</button>
+                                            :
+                                            <button onClick={handleSubmit} disabled className="bg-[#96d600] opacity-50 cursor-not-allowed rounded-md p-2 pl-4 pr-4">Create rule</button>
+                                        }
+                                        
+                                    </div>
+                                </div>
+                            </div>
+                            <div className="w-[60%] text-lg flex flex-col m-auto">
+                                <div className="flex w-full h-[8%] items-center place-items-center">
+                                    <span className="font-medium text-3xl pb-1">Template Preview:</span>
+                                </div>
+                                <div className="aspect-video relative border-4 border-gray-300 rounded-md">
+                                    {selectedTemplateId !== null && templates[selectedTemplateId - 1].templateWidgets.map((templateWidget, index) => 
+                                    <div key={templateWidget.id} className={`absolute`}
+                                        style={{
+                                            width: `${templateWidget.width}%`,
+                                            height: `${templateWidget.height}%`,
+                                            top: `${templateWidget.top}%`,
+                                            left: `${templateWidget.leftPosition}%`
+                                        }}> 
+                                        <div className={`h-full w-full absolute flex flex-col items-center place-content-center border-2 rounded-sm ${selectedColors[index]}`}>
+                                            {contentElement(templateWidget)}
+                                        </div>
+                                    </div>
+                                    )}
+                                    {showContentsPortal && <ScheduleContentModal
+                                        showContentsPortal={showContentsPortal} 
+                                        setShowContentsPortal={setShowContentsPortal}
+                                        widgetId={selectedWidgetId}
+                                        contents={selectedContent}
+                                        setContents={setSelectedContent} />
+                                    }
```

Comment:
tens o boolean fora e dentro do componente nao seria so necessario aqui fora?

Reply:
Penso que somente tendo isso lá dentro força o load do modal, por isso irei tirar o que está lá dentro e irei deixar o de fora. Mas vou testar isso.